platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ImportMigrationConfigurationAdminEntry.java

```diff
@@ -0,0 +1,181 @@
+                        "Importing configuration for [{}] from [{}]; updating existing configuration...",
+                        memoryConfiguration.getPid(),
+                        getPath());
+            } else {
+                if (LOGGER.isDebugEnabled()) {
+                    if (isManagedServiceFactory()) {
+                        LOGGER.debug(
+                                "Importing configuration for [{}-?] from [{}]; creating new factory configuration...",
+                                factoryPid,
+                                getPath());
+                    } else {
+                        LOGGER.debug(
+                                "Importing configuration for [{}] from [{}]; creating new configuration...",
+                                pid,
+                                getPath());
+                    }
+                }
+                try {
+                    cfg = context.createConfiguration(this);
+                } catch (IOException e) {
+                    if (isManagedServiceFactory()) {
+                        getReport().record(new MigrationException(
+                                "Import error: failed to create factory configuration [%s] with factory pid [%s]; %s.",
+                                getPath(),
+                                factoryPid,
+                                e));
+                    } else {
+                        getReport().record(new MigrationException(
+                                "Import error: failed to create configuration [%s] with pid [%s]; %s.",
+                                getPath(),
+                                pid,
+                                e));
+                    }
+                    return false;
+                }
+                LOGGER.debug(
+                        "Importing configuration for [{}] from [{}]; initializing configuration...",
+                        cfg.getPid(),
+                        getPath());
+            }
+            try {
+                cfg.update(properties);
+                this.restored = true;
+            } catch (IOException e) {
+                getReport().record(new MigrationException(
+                        "Import error: failed to update configuration [%s] with pid [%s]; %s.",
+                        getPath(),
+                        getPid(),
+                        e));
+            }
+        }
+        return restored;
+    }
+
+    public String getFactoryPid() {
+        return factoryPid;
+    }
+
+    public String getPid() {
+        return pid;
+    }
+
+    public boolean isManagedServiceFactory() {
+        return factoryPid != null;
+    }
+
+    public boolean isManagedService() {
+        return factoryPid == null;
+    }
+
+    public Dictionary<String, Object> getProperties() {
+        return properties;
+    }
+
+    private boolean propertiesMatch() {
+        final Dictionary<String, Object> memprops = memoryConfiguration.getProperties();
```

Comment:
❓ Couldn't `memoryConfiguration` be null?

Reply:
It is only called internally after in a condition that ensures it ain't null